rules/key_processor.go

```diff
@@ -72,8 +69,11 @@ func (v3kp *v3KeyProcessor) dispatchWork(index int, rule staticRule, logger *zap
 		ruleIndex:        index,
 		ruleTask:         task,
 		ruleTaskCallback: v3kp.callbacks[index],
-		metricsInfo:      newMetricsInfo(context, keyPattern),
```

Comment:
Humm, should not really move this one as it pulls the method name from the context and starts the clock which is used for the `WorkerQueueWaitTime` metric.

Reply:
I adjusted it so that the time is still started at that point, but then passed into the metrics info when the context is obtained